### PR TITLE
Update infrastructure 20200126

### PR DESCRIPTION
**Pull request by phil-davis**
1) require PHP 7.1 or later and phpunit 7
2) test on PHP 7.1 through 7.4
3) switch to "standard" php-cs-fixer for code style
4) add `phpstan.neon` and run in CI at level 2 (which passes)

### CODE REVIEW

File: phpstan.neon [hunk 1]
```diff
@@ -0,0 +1,3 @@
+parameters:
+  level: 0
+  bootstrap: %currentWorkingDirectory%/vendor/autoload.php
```
Comment (staabm) [inline phpstan.neon]: Afaik the default composer  autoloader dont need to be configured

**Review by staabm — APPROVED** [review]
Lgtm - except the few nits

File: composer.json [hunk 2]
```diff
@@ -31,8 +31,8 @@
         }
     },
     "require-dev": {
-        "sabre/cs": "^1.0",
-        "phpunit/phpunit" : "*"
+        "friendsofphp/php-cs-fixer": "^2",
```
Comment (staabm) [inline composer.json]: As mentioned in the other PR, we should add the minor for the cs-fixer

File: phpstan.neon (hunk 1 above)
Comment (phil-davis) [reply phpstan.neon]: OK, I copied this file from other repos.
I will try without this line.

Comment (phil-davis) [reply phpstan.neon]: You are correct - that line is not needed. Removed it.

File: composer.json (hunk 2 above)
Comment (phil-davis) [reply composer.json]: Pinned it to 2.16.*

**Comment by phil-davis** [comment]
@staabm ready for merge

**Comment by phil-davis** [comment]
@staabm I notice that some repos have `phpunit.xml` and some have `phpunit.xml.dist`
It would be nice to be consistent. Which one should I make the "standard"?

**Comment by staabm** [comment]
phpunit.xml

Feel free to merge

**Comment by staabm** [comment]
Just updated permissions for most repos

**Comment by phil-davis** [comment]
@staabm I don't have a merge button. Please review and merge.

**Comment by staabm** [comment]
Great job, thx!

**Pull request merged**